crates/ffmpeg/src/io/channel.rs

```diff
@@ -181,3 +181,255 @@ impl<T: ChannelCompatSend> std::io::Write for ChannelCompat<T> {
         Ok(())
     }
 }
+
+#[cfg(test)]
+#[cfg_attr(all(test, coverage_nightly), coverage(off))]
+mod tests {
+    use std::io::{Read, Write};
+    use std::sync::mpsc;
+    use std::thread;
+
+    use tokio::sync::broadcast;
+    use tokio::sync::mpsc::unbounded_channel;
+
+    use crate::io::channel::{ChannelCompat, ChannelCompatRecv, ChannelCompatSend};
+
+    #[test]
+    fn test_channel_compat_new() {
+        let (tx, _rx) = mpsc::channel::<Vec<u8>>();
+        let channel_compat = ChannelCompat::new(tx);
+
+        assert!(
+            channel_compat.inner.lock().is_ok(),
+            "Inner Mutex should be initialized and lockable"
+        );
+        assert_eq!(channel_compat.total, 0, "Total should be initialized to 0");
+        assert_eq!(channel_compat.pkt_idx, 0, "pkt_idx should be initialized to 0");
+        assert!(channel_compat.buffer.is_empty(), "Buffer should be empty at initialization");
+    }
+
+    #[test]
+    fn test_into_compat() {
+        struct MockReceiver {
+            data: Option<Vec<u8>>,
+        }
+
+        impl ChannelCompatRecv for MockReceiver {
+            type Data = Vec<u8>;
+
+            fn channel_recv(&mut self) -> Option<Self::Data> {
+                self.data.take()
+            }
+        }
+
+        let receiver = MockReceiver {
+            data: Some(vec![1, 2, 3]),
+        };
+        let compat = receiver.into_compat();
+
+        assert!(compat.inner.lock().is_ok(), "Inner Mutex should be initialized and lockable");
+        assert_eq!(compat.total, 0, "Total should be initialized to 0");
+        assert_eq!(compat.pkt_idx, 0, "pkt_idx should be initialized to 0");
+        assert!(compat.buffer.is_empty(), "Buffer should be empty at initialization");
+    }
+
+    #[test]
+    fn test_into_compat_send() {
+        struct MockSender {
+            sent_data: Vec<Vec<u8>>,
+        }
+
+        impl ChannelCompatSend for MockSender {
+            type Data = Vec<u8>;
+
+            fn channel_send(&mut self, data: Self::Data) -> bool {
+                self.sent_data.push(data);
+                true
+            }
+        }
+
+        let sender = MockSender { sent_data: vec![] };
+        let compat = sender.into_compat();
+
+        assert!(compat.inner.lock().is_ok(), "Inner Mutex should be initialized and lockable");
+        assert_eq!(compat.total, 0, "Total should be initialized to 0");
+        assert_eq!(compat.pkt_idx, 0, "pkt_idx should be initialized to 0");
+        assert!(compat.buffer.is_empty(), "Buffer should be empty at initialization");
+    }
+
+    #[test]
+    fn test_mpsc_channel_send_recv() {
```

Comment:
This test needs a feature flag so that it does not compile if tokio is not enabled.